build fails on macOS using GNU sed

On macs that use the GNU tooling over BSD, the build fails with the following error. The date error below seems to be a symptom of the sed issue.

```
% ./build/run make -j4                                                                                     17-11-17 - 11:18:40
==== building the cross container (this could take minutes the first time)
=== cross build image failed for build-f18d7920/cross-amd64
date: invalid option -- 'v'
Try 'date --help' for more information.
date: invalid option -- 'v'
Try 'date --help' for more information.
=== docker build build-f18d7920/rootfs-builder-amd64
sed: can't read s|BASEIMAGE|ubuntu:zesty|g: No such file or directory
make[1]: *** [Makefile:36: rootfs-build] Error 2
make: *** [Makefile:27: base.linux_amd64] Error 2
```

--- a/build/makelib/common.mk
+++ b/build/makelib/common.mk
@@ -64,13 +64,7 @@ CXX := $(CROSS_TRIPLE)-g++
 export CC CXX
 endif
 
-UNAME_S:=$(shell uname -s)
-ifeq ($(UNAME_S),Darwin)
-SED_CMD?=sed -i ""
-endif
-ifeq ($(UNAME_S),Linux)
-SED_CMD?=sed -i
-endif
+SED_CMD?=sed -i -e
 
 # set the version number. you should not need to do this
 # for the majority of scenarios.

--- a/images/image.mk
+++ b/images/image.mk
@@ -127,12 +127,7 @@ PRUNE_KEEP ?= 24
 PRUNE_DRYRUN ?= 0
 
 CACHE_DATE_FORMAT := "%Y-%m-%d.%H%M%S"
-ifeq ($(UNAME_S),Linux)
-CACHE_PRUNE_DATE := $(shell date -u --date="$(PRUNE_HOURS) hours ago" +"$(CACHE_DATE_FORMAT)")
-endif
-ifeq ($(UNAME_S),Darwin)
-CACHE_PRUNE_DATE := $(shell date -u -v -$(PRUNE_HOURS)H +"$(CACHE_DATE_FORMAT)")
-endif
+CACHE_PRUNE_DATE := $(shell export TZ="UTC+$(PRUNE_HOURS)"; date +"$(CACHE_DATE_FORMAT)")
 CACHE_TAG := $(shell date -u +"$(CACHE_DATE_FORMAT)")
 
 cache.lookup:

